src/Maestro/DependencyUpdateErrorProcessor/DependencyUpdateErrorProcessor.cs

```diff
@@ -0,0 +1,485 @@
+        /// <returns></returns>
+        private async Task CreateOrUpdateGithubIssueAsync(RepositoryBranchUpdateHistoryEntry updateHistoryError, 
+            string issueRepo, Func<string, string, bool> shouldReplaceDescription, string description)
+        {
+            _logger.LogInformation($"Error Message : '{updateHistoryError.ErrorMessage}' in repository :  '{updateHistoryError.Repository}'");
+            IReliableDictionary<(string repository, string branch),int> gitHubIssueEvaluator =
+                await _stateManager.GetOrAddAsync<IReliableDictionary<(string repository, string branch), int>>("gitHubIssueEvaluator");
+            GitHubClient client = await AuthenticateGitHubClient(issueRepo);
+            var parseRepoUri = ParseRepoUri(issueRepo);
+            Octokit.Repository repo = await client.Repository.Get(parseRepoUri.owner, parseRepoUri.repo);
+            var issueNumber = new ConditionalValue<int>();
+            using (ITransaction tx = _stateManager.CreateTransaction())
+            {
+                issueNumber = await gitHubIssueEvaluator.TryGetValueAsync(tx,
+                    (updateHistoryError.Repository, updateHistoryError.Branch));
+                await tx.CommitAsync();
+            }
+
+            if (issueNumber.HasValue)
+            {
+                Issue issue = await client.Issue.Get(repo.Id, issueNumber.Value);
+                // check if the issue is open only then update it else create a new issue and update the dictionary.
+                if (issue.State.Equals("Open"))
+                {
+                    _logger.LogInformation(
+                        $@"Updating a gitHub issue number : '{issueNumber}' for the error : '{updateHistoryError.ErrorMessage}'
+                         for the repository : '{updateHistoryError.Repository}'");
+                    await UpdateIssueAsync(client, updateHistoryError, shouldReplaceDescription,
+                        description, issue, repo.Id);
+                    return;
+                }
+            }
+            // Create a new issue for the error if the issue is already closed or the issue does not exists.
+                _logger.LogInformation(
+                    $@"Creating a new gitHub issue for dependency Update Error, 
+                    for the error message  '{updateHistoryError.ErrorMessage}'
+                    for the repository : '{updateHistoryError.Repository}'");
+                await CreateIssueAsync(client, updateHistoryError, gitHubIssueEvaluator, description, repo.Id,
+                    issueRepo);
+        }
+
+        /// <summary>
+        ///     Parse out the owner and repo from a repository url
+        /// </summary>
+        /// <param name="uri">Github repository URL</param>
+        /// <returns>Tuple of owner and repo</returns>
+        public static (string owner, string repo) ParseRepoUri(string uri)
+        {
+            var u = new UriBuilder(uri);
```

Comment:
Why UriBuilder and not just "Uri"? We don't appear to be "building" anything here.